core/test/forest/LocalLinearForestTest.cpp

```diff
@@ -0,0 +1,66 @@
+/*-------------------------------------------------------------------------------
+  This file is part of generalized random forest (grf).
+
+  grf is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  grf is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with grf. If not, see <http://www.gnu.org/licenses/>.
+ #-------------------------------------------------------------------------------*/
+
+#include "commons/utility.h"
+#include "forest/ForestPredictor.h"
+#include "forest/ForestPredictors.h"
+#include "forest/ForestTrainer.h"
+#include "forest/ForestTrainers.h"
+#include "utilities/ForestTestUtilities.h"
+
+#include "catch.hpp"
+
+
+TEST_CASE("LLF predictions vary linearly with Y", "[local_linear, forest]") {
+    // Run the original forest.
+
+    Data* data = load_data("test/forest/resources/gaussian_data.csv");
```

Comment:
Would it be possible to use a smaller dataset here? The tests takes quite a while to run. You could probably just copy the current dataset and cut it down to 50 samples.